src/crimson/osd/pg_backend.cc

```diff
@@ -215,6 +216,101 @@ PGBackend::read(const object_info_t& oi,
     });
 }
 
+namespace {
+
+  template<class CSum>
+  void do_checksum(ceph::bufferlist& init_value_bl,
+		   size_t chunk_size,
+		   const ceph::bufferlist& buf,
+		   ceph::bufferlist& result)
+  {
+    typename CSum::init_value_t init_value;
+    auto init_value_p = init_value_bl.cbegin();;
+    try {
+      decode(init_value, init_value_p);
+      // chop off the consumed part
+      init_value_bl.splice(0, init_value_p.get_off());
+    } catch (const ceph::buffer::end_of_buffer&) {
+      logger().warn("{}: init value not provided", __func__);
+      throw crimson::osd::invalid_argument();
```

Comment:
Hmm, this would require a bit more effort.

```suggestion
      return crimson::ct_error::invarg::make()
```